samples/apps/pipes.jakt

```diff
@@ -0,0 +1,183 @@
+
+enum Stream : c_int {
+    STDIN  = 0,
+    STDOUT = 1,
+    STDERR = 2
+}
+
+struct Pipe {
+    reading_end: c_int
+    writing_end: c_int
+}
+
+class System {
+    public function getErrno() -> c_int {
+        unsafe { cpp {
+            "return errno;"
+        }}
+        // Unreachable code
+        return -1
+    }
+
+    public function pipe() throws -> Pipe  {
+        unsafe{
+            let mutable pipe_ret : c_int = 0;
+            let mutable fds : [c_int] = [0 as! c_int, 0 as! c_int]
+            cpp {
+                "pipe_ret = ::pipe(fds.unsafe_data());"
+            }
+
+            if pipe_ret == 0 {
+                return Pipe(reading_end: fds[0], writing_end: fds[1])
+            } else {
+                throw Error::from_errno(getErrno() as! i32)
+            }
+
+            // Unreachable code
+            return Pipe(reading_end: 0, writing_end: 1)
+        }
+    }
+
+    public function dup2(replace: c_int, with: c_int) throws -> c_int {
+        unsafe{
+            let mutable dup2_ret : c_int = 0;
+
+            cpp {
+                "dup2_ret = ::dup2(with, replace);"
+            }
+
+            if dup2_ret >= 0 {
+                return dup2_ret
+            } else {
+                throw Error::from_errno(getErrno() as! i32)
+            }
+
+            // Unreachable code
+            return -1
+        }
+    }
+
+    public function fork() throws -> c_int {
+        unsafe{
+            let mutable fork_ret : c_int = 0;
+
+            cpp {
+                "fork_ret = ::fork();"
+            }
+
+            if fork_ret as! i64 == -1 {
```

Comment:
Do we have to cast this? `dup2_ret >= 0` works...

Reply:
Yes, for now (#465)